lake-primitives/src/types/events.rs

```diff
@@ -26,6 +30,19 @@ impl Event {
     pub fn related_receipt_id(&self) -> &crate::CryptoHash {
         &self.related_receipt_id
     }
+
+    pub fn related_receipt_receiver_id(&self) -> &AccountId {
+        &self.receiver_id
+    }
+
+    pub fn related_receipt_predecessor_id(&self) -> &AccountId {
+        &self.predecessor_id
+    }
+
+    // checks it predecessor_id or receiver_id is equal to the given account_id
+    pub fn is_related_to(&self, account_id: &AccountId) -> bool {
+        &self.receiver_id == account_id || &self.predecessor_id == account_id
+    }
```

Comment:
This is misleading and its turn the `events_by_account_id` is also misleading. Besides the predecessor account (sender), and receiver account (contract), there are often some event-specific accounts involved (e.g. when `nft_transfer` happens, there are `old_owner_id` and `new_owner_id`.

I think you should only provide a helper to filter by contract account id (receiver account id)